dali/pipeline/operators/util/pad.cu

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <map>
+#include "dali/pipeline/operators/util/pad.h"
+#include "dali/core/static_switch.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+namespace detail {
+
+template <typename Type>
+void RunHelper(TensorList<GPUBackend> &output,
+               const TensorList<GPUBackend> &input,
+               Type padding_val,
+               int axis,
+               cudaStream_t stream,
+               kernels::ScratchpadAllocator &scratch_alloc) {
+  std::size_t number_of_dims = input.tensor_shape(0).size();
+  VALUE_SWITCH(number_of_dims, NumDims, (1, 2, 3, 4), (
+    kernels::PadGPU<Type, NumDims> kernel;
+    kernels::KernelContext ctx;
+    ctx.gpu.stream = stream;
+    auto in_view = view<const Type, NumDims>(input);
+
+    kernels::KernelRequirements req = kernel.Setup(ctx, in_view, axis);
+
+    output.set_type(TypeInfo::Create<Type>());
+    output.Resize(req.output_shapes[0]);
+
+    scratch_alloc.Reserve(req.scratch_sizes);
+    auto scratchpad = scratch_alloc.GetScratchpad();
+    ctx.scratchpad = &scratchpad;
+
+    auto out_view = view<Type, NumDims>(output);
+    // TODO(spanev): add per sample padding val
```

Comment:
do you plan to add it in this PR?

Reply:
It is described in schema so I guest that he does.

Reply:
Not in this PR,I don't think it strongly needed